testsuite/features/support/setup_browser.rb

```diff
@@ -55,6 +55,13 @@ def enable_assertions
   end
 end
 
+AfterStep do
+  if all('.senna-loading-bar', visible: true).any?
+    puts "Step ends with an ajax transition not finished, let's wait a bit!"
+    raise 'Timeout: Waiting AJAX transition' unless has_no_css?('.senna-loading')
+  end
+end
```

Comment:
uuh this is extreme! ahah

Do we still need all the wrappers with this AfterStep logic?

Reply:
The most import of that piece of code is that `all('.senna-loading-bar', visible: true).any?` is instantaneous, this method has not any kind of implicit `wait`. So, in case it doesn't exist, it will not waste time waiting for it.
Second, I just noticed it is wrong, as it should be `all('.senna-loading').any?` instead, as we agreed this morning, I will change.
And.. about the wrappers, honestly, still not sure what's the best way to proceed, even with the wrappers after the full test suite, we had 17 cases where this AfterStep condition was true, so the wrappers are not properly working some times, my guess is that senna script is slower than capybara, so after the click, when we check if this element is not on the page, in fact is not because still it was not loaded and the ajax transition still not started.
Even with this afterstep, it could be the case where the transition still not started :/ 
The only solution I can see to fix it, it would pass throw a change in the product code.... adding a timestamp into an element on the page, so in the wrappers of clicking, we wait and check that the page actually has a timestamp newer than the moment we clicked, and only when this condition is satisfied we can continue to the next step. In that case... we can rid of that AfterStep extreme thingy ^^o